domain/operation/service/action.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	coreaction "github.com/juju/juju/core/action"
+	"github.com/juju/juju/domain/operation"
+)
+
+// GetAction returns the action identified by its UUID.
+func (s *Service) GetAction(ctx context.Context, actionUUID coreaction.UUID) (operation.Action, error) {
```

Comment:
Don't forget to validate the UUID here in a follow up PR.

Same for CancelAction.